tests/integration/targets/win_domain_ou/tasks/check_mode_test.yml

```diff
@@ -0,0 +1,87 @@
+---
+- name: Run Check Mode Tests
+  block:
+    - name: Ensure OU is present (check_mode)
+      community.windows.win_domain_ou:
+        name: ansible_checkmode
+      register: test1c
+      failed_when: test1c is changed
```

Comment:
This should be `changed`. A module reports a change in check mode if it was going to make a change (even though it didn't). This leads into one of my comments above where the changed result should be set in the places you call a cmdlet that makes a change rather than do it on the result.

Reply:
Thank you misunderstood how this works updated and fixed